04_Intro_to_Butler.ipynb

```diff
@@ -426,39 +473,58 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "ra = 70.37770\n",
-    "dec = -37.1757\n",
-    "s1 = \"exposure.day_obs = 20251201\"\n",
-    "s2 = \"exposure.tracking_ra > \"+str(ra-1.0)\n",
-    "s3 = \"exposure.tracking_ra < \"+str(ra+1.0)\n",
-    "s4 = \"exposure.tracking_dec > \"+str(dec-1.0)\n",
-    "s5 = \"exposure.tracking_dec < \"+str(dec+1.0)\n",
-    "\n",
-    "results = registry.queryDimensionRecords('visit',\n",
-    "                                         where=s1+\" AND \"+s2+\" AND \"+s3+\" AND \"+s4+\" AND \"+s5,\n",
-    "                                         collections=collection)\n",
+    "import lsst.sphgeom\n",
     "\n",
-    "# Use expandDataId to fill in the implicit dataId keys with values\n",
-    "for i, ref in enumerate(results):\n",
-    "    tempId = butler.registry.expandDataId(ref.dataId)\n",
-    "    print(tempId.full)\n",
-    "    if i > 10:\n",
+    "pixelization = lsst.sphgeom.HtmPixelization(7)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "htm_id = pixelization.index(\n",
+    "    lsst.sphgeom.UnitVector3d(\n",
+    "        lsst.sphgeom.LonLat.fromDegrees(70.37699524983329, -37.17573628348882)\n",
+    "    )\n",
+    ")\n",
+    "scale = pixelization.triangle(htm_id).getBoundingCircle().getOpeningAngle().asDegrees()*3600\n",
+    "print(f'HTM ID={htm_id} at level={pixelization.getLevel()} is a ~{scale:0.2}\" triangle.')"
```

Comment:
I am somewhat inclined to recommend using this instead of the `findTract` stuff in later sections - once you have an HTM ID, you can use that to query for tract+patch data products like coadds directly, and I think `findTract` may have exhibit some surprising behavior as well as being slow.  But this approach is (as noted below) also slow and non-ideal, so I went with the minimal change.  You're welcome to turn this into a function and use it later in the notebook if you like.

Reply:
Does the `skymap` package have a routine for converting an ra/dec to a tract/patch that could then be used in the query?  All this sphgeom stuff seems like a lot of complex code for something that is going to be common.

Reply:
It does - that's the `findTract` I'm referring to.  I think sphgeom approach is probably closer to the implementation we want, and certainly has better behavior (`findTract` will tell you the tract whose _center_ is closest to your point, not the tract that actually overlaps it).  But the sphgeom approach is not wrapped up in a friendly interface at all, and it may well be slower than `findTract` because it does have to talk to the database; what we really want is a sphgeom-backed implementation of skymap.